tests/integration/test_cloudformation.py

```diff
@@ -2334,134 +1647,125 @@ def test_cfn_with_apigateway_resources(self):
-        assert "StackId" in rs
-        assert stack_name in rs["StackId"]
-
-        list_functions = list_all_resources(
-            lambda kwargs: lambda_client.list_functions(**kwargs),
-            last_token_attr_name="nextToken",
-            list_attr_name="Functions",
-        )
-        list_roles = list_all_resources(
-            lambda kwargs: iam.list_roles(**kwargs),
-            last_token_attr_name="nextToken",
-            list_attr_name="Roles",
-        )
-
-        new_function = [
-            function
-            for function in list_functions
-            if function.get("FunctionName") == lambda_function_name
-        ]
-        new_role = [role for role in list_roles if role.get("RoleName") == lambda_role_name]
-
-        assert len(new_function) == 1
-        assert lambda_role_name in new_function[0].get("Role")
-
-        assert len(new_role) == 1
-
-        # Generate new names for lambda and IAM Role
-        lambda_role_name_new = "lambda-role-%s" % short_uid()
-        lambda_function_name_new = "lambda-function-%s" % short_uid()
-
-        template["Resources"]["LambdaExecutionRole"]["Properties"][
-            "RoleName"
-        ] = lambda_role_name_new
-        template["Resources"]["LambdaFunction1"]["Properties"][
-            "FunctionName"
-        ] = lambda_function_name_new
-
-        # Update stack and wait for 'UPDATE_COMPLETE' status of the stack
-        rs = update_and_await_stack(stack_name, TemplateBody=json.dumps(template))
-
-        # Checking new required values for Lambda function and IAM Role
-        assert "StackId" in rs
-        assert stack_name in rs["StackId"]
-
-        list_functions = list_all_resources(
-            lambda kwargs: lambda_client.list_functions(**kwargs),
-            last_token_attr_name="nextToken",
-            list_attr_name="Functions",
-        )
-
-        list_roles = list_all_resources(
-            lambda kwargs: iam.list_roles(**kwargs),
-            last_token_attr_name="nextToken",
-            list_attr_name="Roles",
-        )
-
-        new_function = [
-            function
-            for function in list_functions
-            if function.get("FunctionName") == lambda_function_name_new
-        ]
-        assert len(new_function) == 1
-        assert lambda_role_name_new in new_function[0].get("Role")
-        new_role = [role for role in list_roles if role.get("RoleName") == lambda_role_name_new]
-        assert len(new_role) == 1
-
-        # Delete the stack and wait for the status 'DELETE_COMPLETE' of the stack
-        delete_and_await_stack(stack_name)
-
-    def test_cfn_with_multiple_route_tables(self):
-        ec2_client = aws_stack.create_external_boto_client("ec2")
-
+    # TODO: this test seems a bit convoluted without a clear goal
+    # def test_updating_stack_with_iam_role(self):
+    #     lambda_client = aws_stack.create_external_boto_client("lambda")
+    #     iam = aws_stack.create_external_boto_client("iam")
```

Comment:
i think we should either remove the test all together if it adds no value or is covered by other tests, or simply skip it if we want to keep the code for some reason. we shouldn't keep code in the repo that commented out where it's not clear what to do with it.

Reply:
Added a slightly refactored version back but kept a todo to re-evaluate. There a few tests in there that we might be able to drop soon :tm: